content/articles/2017-12-workflow-vs-script.md

```diff
@@ -0,0 +1,126 @@
+
+The here package piggybacks on the Project setup and uses the presence of an `.Rproj` file to recognize the top-level of your project. Last but not least, you can double click on the `.Rproj` file to launch an instance of RStudio where the associated R process and file browser are conveniently pointed at the correct folder. RStudio fully supports Project-based workflows, making it easy to switch from one to another, have many open at once, re-launch recently used Projects, etc.
+
+Read up on the [here package](https://CRAN.R-project.org/package=here) to learn about more features, such as additional ways to mark the top directory and troubleshooting with `dr_here()`. I have also written a [more detailed paean](https://github.com/jennybc/here_here) about this package before.
+
+## What's wrong with `rm(list = ls())`?
+
+It's also fairly common to see data analysis scripts that begin with this workspace-nuking command:
+
+``` r
+rm(list = ls())
+```
+
+Just like hard-wiring the working directory, this is, again, highly suggestive the useR works in one R process and manually switches gears when they shift from one project to another. It's clear that "situation normal" for this user is an R process that is used, not fresh and clean.
+
+The problem is that `rm(list = ls())` does NOT, in fact, create a fresh R process. All it does is delete user-created objects from the global workspace.
+
+Many other important changes to the R landscape persist invisibly. Any packages that have been loaded are still available. Any options that have been set to non-default values remain that way. Working directory is unaffected (which is, of course, why we see `setwd()` so often here too!).
+
+Why does this matter? It means your script is now vulnerable to hidden dependencies on things you ran in this R process before you executed `rm(list = ls())`.
+
+  * You might use functions from a package without including the necessary `library()` call. Your collaborator won't be able to run this script.
+  * You might code up an entire analysis assuming that `stringsAsFactors = FALSE` but next week, when you have restarted R, everything will inexplicably be broken.
+  * You might write paths relative to some random working directory, then be puzzled next month when nothing can be found or results don't appear where you expect.
+
+The solution is to write every script assuming it will be run in a fresh R process. How do you enforce this? Two key steps:
+
+  * User-level setup: Do not save `.RData` when you quit R and don't load `.RData` when you fire up R. In RStudio, this behavior can be requested in the General tab of Preferences.
```

Comment:
I wonder if we should wrap this behaviour up into a function somewhere? RStudio settings are a file with a known structure in a known location so we could either edit ourselves or at least warn the user if not set up.